### DISCUSSION TITLE
Connect by hostname on a Windows network

### DISCUSSION THREAD

**Discussion by Tlem33** [rustdesk/rustdesk#10159] [Q&A]
Hello,
I searched the 40 pages of discussions, without finding a functional answer.

I found these two discussions, but they do not answer my problem.
https://github.com/rustdesk/rustdesk/discussions/8119
https://github.com/rustdesk/rustdesk/discussions/5833

In a classic Windows network, I try to connect from one machine to another using the hostname of the PC, but systematically I have a connection error (ID does not exist).

Details:
On the "server" workstation I have activated the connection by IP (port 21118) and the connection by the IP address works well.

Just in case, I have disabled IP V6 and I tried to enter the hostname of the PC in the input box and I even tried hostname:21118 without success.
The PCs are not in a domain, just a workgroup.
I also tried:
workgroup/hostname
workgroup\hostname
workgroup/hostname:21118
workgroup\hostname:21118
./hostname:21118
.\hostname:21118
but it still doesn't work.

As a last resort, I used a workaround using a batch script that pings the hostname to retrieve the IP address and then I launch the connection by the IP address.

`@Echo Off
:: Hostname to connect
Set PC-Hostname=EM00TWA5

:: Get IPV4 from hostname
For /F "Delims=[] Tokens=2" %%a In ('Ping -4 -n 1 %PC-Hostname% ^| Findstr [') Do Set NetworkIP=%%a

If Exist "%LOCALAPPDATA%\rustdesk\rustdesk.exe" (
	Start "" "%LOCALAPPDATA%\rustdesk\rustdesk.exe" --connect %NetworkIP%
) Else (
	Start "" "%~DP0rustdesk.exe" --connect %NetworkIP%
)`

Do you have a more direct solution because my workaround, although functional, is not necessarily the simplest solution?

Thanks in advance for any help.

**Comment by GastricBoar** [comment]
Following this, i get the same problem; the only way to connect to another PC in my LAN using hostname is "hostname.domain.dom:21118", just writing a hostname would be fantastic.

**Comment by Tlem33** [comment]
Indeed, it is a solution when you are in a domain, but in a network outside the domain, it does not work.

**Comment by Darkxylitol** [comment]
try [dot]local. ex: Desktop-xyz.local:port
